trie/verkle.go

```diff
@@ -239,45 +237,31 @@ func nodeToDBKey(n verkle.VerkleNode) []byte {
 // Commit writes all nodes to the trie's memory database, tracking the internal
 // and external (for account tries) references.
 func (trie *VerkleTrie) Commit(onleaf LeafCallback) (common.Hash, int, error) {
-	flush := make(chan verkle.VerkleNode)
-	flusher := func(n verkle.VerkleNode) {
+	root, ok := trie.root.(*verkle.InternalNode)
+	if !ok {
+		return common.Hash{}, 0, errors.New("unexpected root node type")
+	}
+	nodes, err := root.BatchSerialize()
+	if err != nil {
+		return common.Hash{}, 0, fmt.Errorf("serializing tree nodes: %s", err)
+	}
+
+	for _, node := range nodes {
 		if onleaf != nil {
-			if leaf, isLeaf := n.(*verkle.LeafNode); isLeaf {
-				for i := 0; i < verkle.NodeWidth; i++ {
-					if leaf.Value(i) != nil {
-						comm := n.Commitment().Bytes()
```

Comment:
There was an extra improvement which might be subtle. Now this `Commitment().Bytes()` is also batched. This line was also doing inverses and was kind of "out of the loop" from go-verkle.